src/mudlet.cpp

```diff
@@ -767,6 +768,11 @@ void mudlet::slot_close_profile_requested( int tab )
     mHostDockConsoleMap.remove(pH);
     mHostConsoleMap.remove(pH);
 
+    for( TToolBar* pTB : hostToolBarMap ) {
+        pTB->setAttribute( Qt::WA_DeleteOnClose );
+        pTB->close();
```

Comment:
I think it would be wise - and more robust - to check that `pTB` is not a `nullptr` before dereferencing (using) it in these two lines.

Having set the `deleteOnClose` flag in the prior line you can get the same behaviour - but in a safer manner - with a `pTB->deleteLater()` call - assuming that `pTB` is not a `nullptr` a widget can only safely receive one `close()` call as it is destroyed (supposedly) immediately and afterwards another attempt to do so will cause a _seg. fault_ - whereas it is safe for a widget to get multiple `deleteLater()` calls which will "optimise" (I think) the process to do so when "convenient" - the Qt Documentation says:
> `[slot] void QObject::deleteLater()`
> Schedules this object for deletion.

> The object will be deleted when control returns to the event loop. If the event loop is not running when this function is called (e.g. `deleteLater()` is called on an object before `QCoreApplication::exec()`), the object will be deleted once the event loop is started. If `deleteLater()` is called after the main event loop has stopped, the object will not be deleted. Since Qt 4.8, if `deleteLater()` is called on an object that lives in a thread with no running event loop, the object will be destroyed when the thread finishes.

> Note that entering and leaving a new event loop (e.g., by opening a modal dialog) will not perform the deferred deletion; for the object to be deleted, the control must return to the event loop from which `deleteLater()` was called.

> Note: It is safe to call this function more than once; when the first deferred deletion event is delivered, any pending events for the object are removed from the event queue.

> See also `destroyed()` and `QPointer`.

This would be safer if there is anything else added now or later that might also have destroyed the members of `hostToolBarMap` in the past {well, using that __and__ making it so that the container that `pH->getActionUnit()->getToolBarList()` returns is a `std::list<QPointer<TToolBar>>` rather than the current `std::list<TToolBar*>`}!